packages/@aws-cdk/aws-appmesh/lib/virtual-gateway.ts

```diff
@@ -173,15 +173,26 @@ export class VirtualGateway extends VirtualGatewayBase {
     }
 
     const accessLogging = props.accessLog?.bind(this);
+    const backendDefaults = props.backendDefaults;
 
     const node = new CfnVirtualGateway(this, 'Resource', {
       virtualGatewayName: this.physicalName,
       meshName: this.mesh.meshName,
       spec: {
         listeners: this.listeners.map(listener => listener.listener),
-        backendDefaults: props.backendDefaults !== undefined
-          ? {
-            clientPolicy: props.backendDefaults?.clientPolicy?.bind(this).clientPolicy,
+        backendDefaults: backendDefaults !== undefined ?
+          {
+            clientPolicy: backendDefaults.tlsClientPolicy !== undefined ?
+              {
+                tls: {
+                  ports: backendDefaults.tlsClientPolicy.ports,
+                  enforce: backendDefaults.tlsClientPolicy.enforce,
+                  validation: {
+                    trust: backendDefaults.tlsClientPolicy.tlsValidationContext.trust.bind(this).virtualGatewayClientTlsValidationContextTrust,
+                  },
+                },
+              }
+              : undefined,
```

Comment:
I think you are asking to create method/function to assign `tls` properties. For that case, would you recommend to:

- Create `bind()` class for `TlsClientPolicy` (*this will also require to create a class)
- Create `render()` class for each classes that assign `tls` properties.

Reply:
I would suggest creating a helper function that takes in a `TlsClientPolicy`, and a function that extracts a `trust` property from the argument, something like this:

```ts
function renderTlsPolicy(scope: Construct, tlsClientPolicy: TlsClientPolicy | undefined, extractor: (c: TlsValidationConfig) => any): any {
  return tlsClientPolicy !== undefined
    ?  {
      tls: { 
        ports: tlsClientPolicy.ports, 
        enforce: tlsClientPolicy.enforce, 
        validation: {
          trust: extractor(tlsClientPolicy.trust.bind(scope))
        },
      },
    }
   : undefined,
}
```

Something like this (the types are not great here, but I think the duplication might be worse).

Reply:
Well noted. I will create `render()` helper function for each classes.